graph/telemetry/istio/appender/response_time.go

```diff
@@ -70,15 +71,25 @@ func (a ResponseTimeAppender) appendGraph(trafficMap graph.TrafficMap, namespace
 	responseTimeMap := make(map[string]float64)
 
 	// query prometheus for the responseTime info in three queries:
+	// note - Istio is migrating their latency metric from seconds to milliseconds. We need to support both until
+	//        the 'seconds' variant is removed. That is why we have these complex queries with OR logic.
 	// 1) query for responseTime originating from "unknown" (i.e. the internet)
 	groupBy := "le,source_workload_namespace,source_workload,source_app,source_version,destination_service_namespace,destination_service_name,destination_workload_namespace,destination_workload,destination_app,destination_version"
-	query := fmt.Sprintf(`histogram_quantile(%.2f, sum(rate(%s{reporter="destination",source_workload="unknown",destination_workload_namespace="%v",response_code=~"%s"}[%vs])) by (%s))`,
+	millisQuery := fmt.Sprintf(`histogram_quantile(%.2f, sum(rate(%s{reporter="destination",source_workload="unknown",destination_workload_namespace="%v",response_code=~"%s"}[%vs])) by (%s))`,
+		quantile,
+		"istio_request_duration_milliseconds_bucket",
+		namespace,
+		"2[0-9]{2}|^0$",         // must match success for all expected protocols
+		int(duration.Seconds()), // range duration for the query
+		groupBy)
+	secondsQuery := fmt.Sprintf(`histogram_quantile(%.2f, sum(rate(%s{reporter="destination",source_workload="unknown",destination_workload_namespace="%v",response_code=~"%s"}[%vs])) by (%s))`,
 		quantile,
 		"istio_request_duration_seconds_bucket",
 		namespace,
 		"2[0-9]{2}|^0$",         // must match success for all expected protocols
 		int(duration.Seconds()), // range duration for the query
 		groupBy)
+	query := fmt.Sprintf(`((%s > 0) OR ((%s > 0) * 1000.0))`, millisQuery, secondsQuery)
```

Comment:
Is the query OK? I don't know much of it, is `1000` multiplying the result of `%s > 0` ?
 `(%s > 0) * 1000.0)`

Reply:
yes, it's tricky but it works.  in ((%s > 0) * 1000.0)) the %s is the main query that returns the quantile value in seconds, then we throw out any time-series where value is 0, and then we multiply by 1000 to turn seconds into milliseconds, so the values from both sides of the OR return in ms.